scala/lambdas/preingest-tdr-package-builder/src/main/scala/uk/gov/nationalarchives/preingesttdrpackagebuilder/Lambda.scala

```diff
@@ -234,9 +235,17 @@ class Lambda extends LambdaRunner[Input, Output, Config, Dependencies]:
       .map(_ => new Output(input.batchId, input.groupId, URI.create(s"s3://${config.rawCacheBucket}/${input.batchId}/metadata.json"), input.retryCount, ""))
   }
 
+  private def truncate(s: String) = {
+    @tailrec
+    def truncateArray(arr: Array[String]): Array[String] =
+      if arr.isEmpty || arr.dropRight(1).mkString(" ").length < 100 then arr
+      else truncateArray(arr.dropRight(1))
+    if s.length < 100 then s else s"${truncateArray(s.split(" ")).mkString(" ")}..."
```

Comment:
What happens if the string is over 100 chars but doesn't have any spaces e.g. a url, file path, code or the start is delimited with something other than a space? Would the final string end up being just "..." since `else truncateArray(arr.dropRight(1))` will return an empty array? Maybe we should also check if the array is of length 1 and then just slice it at 100 chars if it is?